---
output:
  html_document:
    includes:
      after_body: doc_suffix.html
      before_body: doc_prefix.html
      in_header: header.html
    theme: null
    toc: yes
---


```{r echo=FALSE, warning=FALSE, message=FALSE, fig.width=12, fig.height=10}
suppressPackageStartupMessages(library(ggplot2))
suppressPackageStartupMessages(library(knitr))
suppressPackageStartupMessages(library(data.table))

args = c();
#args[1] = "C:/Users/q016hw/Desktop/Projects/mtdna/job-20160426-085751/raw/raw.txt";
#args[2] = "C:/Users/q016hw/Desktop/Projects/mtdna/job-20160426-085751/heteroplasmies/heteroplasmies.txt";
#args[3] = "C:/Users/q016hw/Desktop/Projects/mtdna/job-20160426-085751/haplogroups/haplogroups.txt";
#args[4] = "C:/Users/q016hw/Desktop/Projects/mtdna/job-20160426-085751/haplogroupsCheck/haplogroupsCheck.txt";
#args[5] = "C:/Users/q016hw/Desktop/Projects/mtdna/job-20160426-085751/summary/summary.txt";
#args[6] = "C:/Users/q016hw/Desktop/Projects/mtdna/job-20160426-085751/variants/variants.txt";

args <- commandArgs(trailingOnly = TRUE)
raw = read.table(args[1], header = TRUE, sep="\t");
hetero = read.table(args[2], header = TRUE, sep="\t");
haplogroups<-subset(read.delim(file = args[3], header=FALSE, quote="", stringsAsFactors=FALSE, as.is = TRUE)[1:4], select = c(V1, V3, V4))
haploInfo <- read.table(args[3], header = TRUE, quote="", stringsAsFactors=FALSE, sep="\t");
contamination<-subset(read.delim(file = args[4], header=FALSE, quote="", stringsAsFactors=FALSE, as.is = TRUE)[1:4], select = c(V1,V3, V4))
summary<-read.table(args[5], header = TRUE, sep="\t");
variants = read.table(args[6], quote="", stringsAsFactors=FALSE, header = TRUE, sep="\t");
```

----

## Detected Heteroplasmies 


This interactive table shows the detected heteroplasmic sites per sample. The heteroplasmy level (HET.LEVEL) always displays the percentage of the minor component.

```{r echo=FALSE, results='asis'}
if(nrow(hetero)>0){
	
	hetero$MAJOR.MIN<-with(hetero, paste(hetero[,4],"/",hetero[,5], sep=""))
	hetero$FWD.REV<-with(hetero, paste(hetero[,8],"/",hetero[,9]))

	kable(hetero[,c(1,2,3,11,33,34,10,31,32)], "html", table.attr = "id=\"hetero_table\" class=\"table table-bordered table-striped display responsive nowrap\"",caption = "TYPE: 1 = reliable heteroplasmy; 2 = Heteroplasmy in low complexity region (LCR); 3 = Major/Minor component are swapped for forward and reverse strand") 

} else{

	cat("<div class='alert alert-neutral'>No heteroplasmic sites found!</div>")
	
}
```

----

## Detected Homoplasmies


This interactive table shows the detected variant sites per sample. 

```{r echo=FALSE, results='asis'}
variants$definingHG <- NULL

if(nrow(variants)>0){

	for (i in 1:nrow(variants) ) {
         patternYes = paste(variants[i,2], substr(variants[i,3],3,3), " (yes)", sep="");
         patternHot = paste(variants[i,2], substr(variants[i,3],3,3), " (hotspot)", sep="");
         searchString = as.character(haploInfo[haploInfo$SampleID==variants[i,1],]$Polymorphisms);

         if (length(grep(patternHot, searchString, fixed =TRUE)) >= 1){
            variants$definingHG[i] = "hotspot"
         } else if (length(grep(patternYes, searchString, fixed =TRUE)) >= 1){
            variants$definingHG[i] = "yes"
         } else {
            variants$definingHG[i] = "no"
         }
	}           
	variants$Haplogroup<-paste(variants[,11], "(",variants[,10],")")

	kable(variants[,c(1,2,3,4,5,6,7,8,9,12)], "html", table.attr = "id=\"homo_table\" class=\"table table-bordered table-striped\"")
	 
} else{

	cat("<div class='alert alert-neutral'>No homoplasmic sites found!</div>")
	
}

```

----

## Heteroplasmy Frequencies

This table represents heteroplasmic sites with frequency >= 2. If a position shows up in too many samples, it could be the result of artifacts and needs to be re-checked!

```{r echo=FALSE, results='asis'}

if(nrow(hetero)>0){
	a<-rle(sort(hetero$POS)) 
	b<-data.frame(Position=a$values, Frequency=a$length)
	if (nrow(b[b$Frequency>=2,])>=1){
		kable(b[b$Frequency>=2,], "html", row.names=FALSE, table.attr = "id=\"hetero_freq\" class=\"table table-bordered table-striped\"",caption="")
	}else{
		cat("<div class='alert alert-neutral'>No shared heteroplasmic sites over samples have been detected</div>")
	}
}
```

----

## Classified Haplogroups

Detected haplogroups using <a href="http://haplogrep.uibk.ac.at">HaploGrep</a> based on Phylotree 16. Several coverage values, the average heteroplasmy level (HET.LEVEL) and the amount of detected heteroploasmies (HET.COUNT) are displayed.  


```{r echo=FALSE, results='asis', results='asis'}

if(nrow(haplogroups)>0){
colnames(haplogroups) = haplogroups[1, ]
haplogroups = haplogroups[-1, ]
row.names(haplogroups) <- NULL
haplogroups<-merge(haplogroups, summary, by.x="SampleID", by.y="ID", all.x=TRUE)
kable(haplogroups[, c(1,2,3,4,5,9,10,11,12)], "html", table.attr = "id=\"haplo_table\" class=\"table table-bordered table-striped\"")
} else{
		cat("<div class='alert alert-neutral'>Issues detecting the sample's haplogroup</div>")
	}
```

---

## Haplogroup Potential Contamination / Haplogroup Conflicts

A sample is listed as a <b> possible contamination</b>, if the major and minor haplogroup of the sample differ. 
Always verify the contamination by checking the phylogenetic distance of the two haplogroups using <a href="http://www.phylotree.org/tree/main.html">Phylotree</a>.

```{r echo=FALSE, results='asis'}
if(nrow(contamination)>1){
colnames(contamination) = contamination[1, ]
contamination = contamination[-1, ]
row.names(contamination) <- NULL
df = data.frame(ID = character(), Haplogroup = character(), HaploMajor = character(), HaploMinor = character())
for(i in seq(1, nrow(contamination), 3)){
    if (as.character(contamination[i+2,2])!=as.character(contamination[i+1,2])){
        newRow <- data.frame(SampleId=as.character(contamination[i,1]), Haplogroup=as.character(contamination[i,2]), HaploMajor=as.character(contamination[i+1,2]), HaploMinor=as.character(contamination[i+2,2]))
    	df<-rbind(df, newRow)
    }
}    
if(nrow(df)>0){
	kable(df[,c(1,2,3,4)], "html", table.attr = "id=\"haploCheck_table\" class=\"table table-bordered table-striped\"")
}else{
	cat("<div class='alert alert-success'>no haplogroup based contamination detected</div>")
}

} else{
		cat("<div class='alert alert-neutral'>Issues detecting contamination in the sample's </div>")
	}


```

---


```{r, echo=FALSE, results='asis'}
if(nrow(hetero)>0){
  cat("<hr>");
	cat("<h2>Heteroplasmy Levels per Sample</h2>")
}
```

```{r echo=FALSE, fig.width=12, fig.height=10}
if(nrow(hetero)>0){
	ggplot() + geom_boxplot(data=subset(hetero, TYPE==1), mapping=aes(x=ID, y=HET.LEVEL, color=ID)) + theme(legend.position = "none") + theme(axis.text.x=element_text(angle=+90))
} 
```

```{r, echo=FALSE, results='asis'}
if(nrow(hetero)>0){
	cat("<center>Fig.1: Heteroplasmy level per sample. </center>")
  cat("<hr>");
	cat("<h2>Heteroplasmy Frequencies per Sample</h2>")
}
```

```{r echo=FALSE, fig.width=12, fig.height=10}
count <-as.data.frame(table(hetero$ID));
names(count)[names(count)=="Var1"] <- "Sample";
if(nrow(count)>0){
	ggplot(count, aes(x=Sample, y=Freq)) + geom_bar(fill="cyan4", stat='identity') + theme(axis.text.x=element_text(angle=+90))
}
```

```{r, echo=FALSE, results='asis'}
if(nrow(hetero)>0){
	cat("<center>Fig.2: Heteroplasmic sites per sample.</center>");
	cat("<hr>");
	cat("<h2>Heteroplasmy per Region over all Samples</h2>")
	cat("<p>Amount of heteroplasmic sites grouped according their loci on the mitochondrial genome.</p>")
}
```

```{r echo=FALSE, fig.width=12, fig.height=5, results='asis'}
if(nrow(hetero)>0){
	m1 <- as.vector(hetero$MapLocus)
	m2 <- as.vector(c('MT-DLOOP2','MT-TF','MT-RNR1','MT-TV','MT-RNR2','MT-TL1','MT-ND1','MT-TI','MT-TQ','MT-TM','MT-ND2','MT-TW','MT-TA','MT-TN','MT-TC','MT-TY','MT-CO1','MT-TS1','MT-TD','MT-CO2','MT-TK','MT-ATP8','MT-ATP6','MT-CO3','MT-TG','MT-ND3','MT-TR','MT-ND4L','MT-ND4','MT-TH','MT-TS2','MT-TL2','MT-ND5','MT-ND6','MT-TE','MT-CYB','MT-TT','MT-TP','MT-DLOOP1', ' '))
	mapl<-as.data.frame(table(c(m1,m2)))
	names(mapl)[names(mapl)=="Var1"] <- "Locus"
	mapl$Freq<-mapl$Freq-1
	ggplot() +  geom_bar(data=mapl, mapping=aes(x=Locus, y=Freq, fill=Locus), stat='identity') + theme(axis.text.x=element_text(angle=+90)) + theme(legend.position = "none")
}
```

```{r, echo=FALSE, results='asis'}
if(nrow(hetero)>0){
	cat("<center>Fig.3: Heteroplasmic sites grouped according their loci on the mitochondrial genome. </center>");
}
```

----

## Coverage Plots

The dotted line in purple shows the mean coverage over all analysed samples. The dotted line in turquoise shows the mean coverage for this specific sample.


```{r echo=FALSE, fig.width=5.5, fig.height=4, result='asis'}
rawSelected<- raw[seq(1, nrow(raw), 25),]
rawSelected<-rawSelected[c(1,2,8,9)]
max<-max((rawSelected[3])+(rawSelected[4]))
s<-split(rawSelected, rawSelected$ID)
meanCov<-mean(rawSelected$COV.FWD+rawSelected$COV.REV)
for( i in 1:length(s)) {
	print(ggplot(rbindlist(s[i])) +  geom_line(aes(x= POS, y = COV.FWD+COV.REV), colour="#E69F00")  + geom_hline(aes(yintercept=mean(COV.FWD+COV.REV)), colour="#0072B2", linetype="dashed")+ geom_hline(aes(yintercept=meanCov), colour="#7200B2", linetype="dashed") + xlab("mtDNA position") + ylab("Coverage") + ylim(c(0, max)) +ggtitle(names(s)[i]) )
}

```

<script type="text/javascript">
  $(document).ready(function() {
  	$('#hetero_table').DataTable({
  	
       responsive: {
            details: {
                display: $.fn.dataTable.Responsive.display.modal( {
                    header: function ( row ) {
                        var data = row.data();
                        return 'Details for '+data[0];
                    }
                } ),
                renderer: function ( api, rowIdx, columns ) {
                    var data = $.map( columns, function ( col, i ) {
                        return '<b>' + col.title+':'+'</b><br>'+
                                col.data+'<br><hr>';
                    } ).join('');
 
                    return $('<div/>').append( data );
                }
            }
       },
         
       "fnRowCallback": function( nRow, aData, iDisplayIndex, iDisplayIndexFull ) {
          if ( aData[6] != "1" ){
            $('td', nRow).css('background-color', '#f2dede');
          }
       }
  	
  	});
	} );
</script>

<script type="text/javascript">
  $(document).ready(function() {
  	$('#homo_table').DataTable({
    	"fnRowCallback": function( nRow, aData, iDisplayIndex, iDisplayIndexFull ) {
            if ( aData[7].startsWith("no")){
              $('td', nRow).css('font-weight', 'bold');
            }
         }
  	});
	} );
</script>

<script type="text/javascript">
  $(document).ready(function() {
  	$('#hetero_freq').DataTable();
	} );
</script>

<script type="text/javascript">
  $(document).ready(function() {
  	$('#haplo_table').DataTable();
	} );
	
  $(document).ready(function() {
  	$('#haploCheck_table').DataTable();
	} );	

</script>
